api/src/controllers/login.js

```diff
@@ -303,38 +303,40 @@ const createSessionRetry = (req, retry=10) => {
  * Generates a session cookie for a user identified by supplied token and hash request params.
  * The user's password is reset in the process.
  */
-const loginByToken = (req, res) => {
-  if (!tokenLogin.isTokenLoginEnabled()) {
-    return res.status(400).json({ error: 'disabled', reason: 'Token login disabled' });
-  }
-
+const loginByToken = async (req, res) => {
   if (!req.params || !req.params.token) {
     return res.status(400).json({ error: 'missing', reason: 'Missing required param' });
   }
 
-  return tokenLogin
-    .getUserByToken(req.params.token)
-    .then(userId => {
-      if (!userId) {
-        throw { status: 401, error: 'invalid' };
-      }
+  try {
+    const userId = await tokenLogin.getUserByToken(req.params.token);
+    if (!userId) {
+      return res.status(401).json({ error: 'invalid', reason: 'Invalid token' });
+    }
 
-      return tokenLogin.resetPassword(userId).then(({ user, password }) => {
-        req.body = { user, password, locale: req.body.locale };
+    const userDoc = await users.getUserDoc(userId);
+    if (userDoc.oidc) {
+      return res.status(400).json({ error: 'invalid', reason: 'Token login not allowed for SSO users' });
+    }
```

Comment:
suggestion: calling the `users.getUserDoc` here is adding an additional db round-trip.  Looking at this more closely, I think we could actually just move this validation down into the `tokenLogin.getUserByToken` function (over in the [`shared-libs/user-managment` code](https://github.com/medic/cht-core/blob/d47a578a1ad4fa6b8e59293a79a42ff8fa0cc15e/shared-libs/user-management/src/token-login.js#L286)).  Right before we return the `user._id` we can just check:

```js
      if (user.oidc && config.get().oidc_provider?.client_id) {
  const err = new Error('Token login not allowed for SSO users');
  err.status = 401;
  throw err;
}
```